src/assets/js/main.js

```diff
@@ -85,20 +85,49 @@ function scrollSidebarIntoView() {
   }
 }
 
-function adjustToc() {
-  // Adjustments to the jekyll-toc TOC.
+/**
+ * Adjusts the behavior of the table of contents (TOC) on the page.
+ * 
+ * This function enables a "scrollspy" feature on the TOC, 
+ * where the active link in the TOC is updated
+ * based on the currently visible section in the page.
+ * 
+ * Enables a "back to top" button in the TOC header.
+ */
+function adjustTOC() {
 
   const tocId = '#site-toc--side';
 
   const tocHeader = document.querySelector(tocId + ' header');
 
   if (tocHeader) {
-    tocHeader.addEventListener('click', function (_) {
-      $('html, body').animate({scrollTop: 0}, 'fast');
+    tocHeader.addEventListener('click', (_) => {
+      _scrollToTop();
     });
   }
 
-  $('body').scrollspy({ offset: 100, target: tocId });
+  const tocElement = document.querySelector(tocId);
+  if (tocElement) {
+    window.addEventListener('scroll', _spyOnScroll);
+  }
```

Comment:
I think my last comment may have got lost, but I don't think this change works as intended. It looks like it will add the `active` class to the entire TOC when scrolled past 100px. The intended functionality is to set an individual element as active when its corresponding header (`<h1>`, `<h2`, etc.) is in view.

Since this functionality is from bootstrap through jquery, I think it's fine to keep using it for now: `$('body').scrollspy({ offset: 100, target: tocId });` rather than trying to rewrite that functionality.

We can worry about migrating it if/when we move away from Bootstrap 4 (https://github.com/flutter/website/issues/8030).